test/data-pipeline/data-pipeline-stack.test.ts

```diff
@@ -1222,6 +1246,145 @@ test('Plugins nested stack has CopyAssetsCustomResource', () => {
   });
 });
 
+test('Nested stack has CreateEMRServelsssApplicationCustomResource', () => {
+  const template = nestedTemplates[0];
+  template.hasResourceProperties('AWS::CloudFormation::CustomResource', {
+    projectId: RefAnyValue,
+    name: {
+      'Fn::Join': [
+        '',
+        [
+          'Clickstream-Spark-APP-',
+          RefAnyValue,
+        ],
+      ],
+    },
+    version: RefAnyValue,
+    secourityGroupId: {
+      'Fn::GetAtt': [
+        Match.anyValue(),
+        'GroupId',
+      ],
+    },
+    subnetIds: {
+      'Fn::Join': [
+        ',',
+        {
+          'Fn::Split': [
+            ',',
+            RefAnyValue,
+          ],
+        },
+      ],
+    },
+    idleTimeoutMinutes: RefAnyValue,
+    pipelineS3BucketName: RefAnyValue,
+    pipelineS3Prefix: RefAnyValue,
+  });
+});
+
+test('CreateEMRServelsssApplicationLambdaRole policy is set correctly', () => {
+  const template = nestedTemplates[0];
+  template.hasResourceProperties('AWS::IAM::Policy', {
+    PolicyDocument: {
+      Statement: [
+        {
+          Action: [
+            'logs:CreateLogStream',
+            'logs:PutLogEvents',
+            'logs:CreateLogGroup',
+          ],
+          Effect: 'Allow',
+          Resource: '*',
+        },
+        {
```

Comment:
why putting this custom resource in the vpc?

Reply:
the custom resource  lambda is not put in VPC. 
the lambda need to create an EMR-serverless app which is in VPC, so the lambda need these `ec2: *Interface` permissions.